src/otk/transform.py

```diff
@@ -59,6 +59,11 @@ def resolve_dict(ctx: Context, state: State, tree: dict[str, Any]) -> Any:
     """
 
     for key, val in tree.items():
+        # Replace any variables in a value immediately before doing anything
+        # else, so that variables defined in strings are considered in the
+        # processing of all directives.
+        if isinstance(val, str):
```

Comment:
I wonder if a test for this could be easily extracted from the example given in the commit message for this change?

Reply:
One of the example tests covers this (`09-include-with-var`).  Or would you rather also have a small unit test for it?